packages/ethereum-light-client/src/client_state.rs

```diff
@@ -0,0 +1,39 @@
+use alloy_primitives::{Address, B256, U256};
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+
+use crate::types::{fork_parameters::ForkParameters, height::Height};
+
+#[serde_as]
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug, Default)]
+pub struct ClientState {
+    #[serde_as(as = "DisplayFromStr")]
+    pub chain_id: u64,
+    #[serde(with = "ethereum_utils::base64::fixed_size")]
+    pub genesis_validators_root: B256,
+    pub min_sync_committee_participants: u64, // TODO: Needs be added to e2e tests #143
+    pub genesis_time: u64,
+    pub fork_parameters: ForkParameters,
+    pub seconds_per_slot: u64,
+    pub slots_per_epoch: u64,
+    pub epochs_per_sync_committee_period: u64,
+    pub latest_slot: u64,
+    // TODO: Should this be frozen_slot? Consider this in #143
+    pub frozen_height: Height,
+    #[serde(with = "ethereum_utils::base64::uint256")]
+    pub ibc_commitment_slot: U256,
+    #[serde(with = "ethereum_utils::base64::fixed_size")]
```

Comment:
The reason this, and many of these are here now, is because it is reusing the format we already have. And in that format, the address is bytes, which gets encoded to base64. So if I remove this, the decoding will fail.

The goal was to clean this up and when integrating fully with the e2e tests where new and better types can be formulated. Ideally removing a lot of this kind of stuff.

Reply:
This will be looked at closer in https://github.com/cosmos/solidity-ibc-eureka/issues/143